Incorrect large block size chosen for large files on 32 bit archs

I'm running the test suite for version 0.14.48 on fedora, and it fails on 32 bit architectures with the following error:

```
+ go test [.. snip ..] github.com/syncthing/syncthing/lib/protocol
--- FAIL: TestBlockSize (0.00s)
	protocol_test.go:381: BlockSize(2147483648), size=16777216, expected 2097152
	protocol_test.go:381: BlockSize(3221225472), size=16777216, expected 2097152
FAIL
FAIL	github.com/syncthing/syncthing/lib/protocol	0.457s
```

--- a/lib/protocol/protocol.go
+++ b/lib/protocol/protocol.go
@@ -54,7 +54,7 @@ func init() {
 func BlockSize(fileSize int64) int {
 	var blockSize int
 	for _, blockSize = range BlockSizes {
-		if fileSize < int64(DesiredPerFileBlocks*blockSize) {
+		if fileSize < DesiredPerFileBlocks*int64(blockSize) {
 			break
 		}
 	}

--- a/lib/protocol/protocol_test.go
+++ b/lib/protocol/protocol_test.go
@@ -4,6 +4,7 @@ package protocol
 
 import (
 	"bytes"
+	"encoding/hex"
 	"encoding/json"
 	"errors"
 	"io"
@@ -11,8 +12,6 @@ import (
 	"strings"
 	"testing"
 	"testing/quick"
-
-	"encoding/hex"
 
 	"github.com/syncthing/syncthing/lib/rand"
 )

